Update Skia to 2025-10-06 (ed728afeb3ff52689f260a6dfa23e31df1d93166)

aperezdc opened the pull request:
#### cd69ab5b3082855335376ae79ea50ba0d9d7438e
<pre>
Update Skia to 2025-10-06 (ed728afeb3ff52689f260a6dfa23e31df1d93166)
<a href="https://bugs.webkit.org/show_bug.cgi?id=300228">https://bugs.webkit.org/show_bug.cgi?id=300228</a>

Unreviewed. The changes are localized to Source/ThirdParty/skia.

% git log --oneline --pretty=&apos;%h %s&apos; 953ef34aabaeda010e05c578a8e2de5c08e25c91..ed728afeb3ff52689f260a6dfa23e31df1d93166
ed728afeb3 Roll ANGLE from 62b00f866364 to fc98b3f62a98 (11 revisions)
45191c22b1 Roll Dawn from 01940842b667 to d8f0feef7c17 (14 revisions)
65347fdf4d Roll Skia Infra from 228c951bd699 to b57557664701 (8 revisions)
e7c94f0f38 Roll vulkan-deps from 207fff6486db to 7b14f3ac9637 (1 revision)
1fd0ca1f21 Roll SKP CIPD package from 533 to 534
5479115ef5 Roll vulkan-deps from a45aa5b3843e to 207fff6486db (1 revision)
35f8b9a341 [graphite] Add findOrCreateScratchBuffer to ResourceProvider
67489f2f45 [graphite] add missing include in VelloComputeSteps.h
3d6c568f17 Roll vulkan-deps from edacf5135c8d to a45aa5b3843e (8 revisions)
f316de3d47 Roll recipe dependencies (trivial).
81cab272e3 Return optional raw for path/builder
9cda1a2050 [graphite] Remove assert in VellComputeStep
8a7f88c5b3 [graphite] Move functionality into base BufferWriter
a454242c39 Reland &quot;Reland &quot;[graphite] Extracts early in drawGeometry&quot;&quot;
eb27c53d93 [graphite] Add all blends to NotifyInUseTests
cb9fec0711 Roll vulkan-deps from 7bc87a4a45cf to edacf5135c8d (4 revisions)
b842026480 Roll ANGLE from 7994bf76d7a9 to 62b00f866364 (11 revisions)
3122894ae6 Roll recipe dependencies (trivial).
a256cc54c5 Roll Skia Infra from 8d4953412be2 to 228c951bd699 (11 revisions)
f86ae41132 Roll Dawn from 603a5155599a to 01940842b667 (22 revisions)
1720a85a50 Roll recipe dependencies (trivial).
c6effcb43b Roll vulkan-deps from 3964dd7c126b to 7bc87a4a45cf (7 revisions)
c7792a4a68 Roll recipe dependencies (trivial).
93e2cb6281 Add tests for SkSGMerge
cf339ab390 Make SkAlignTo templated; optimize SkAlignNonPow2
0d6067256c Roll recipe dependencies (trivial).
86f7e448c8 [graphite] Add drawContext pointer to task dumping
7a8c30bd76 use pathbuilder or factories, rather than mutate SkPath
c7f4bad47e Update pixmap asserts to identify negative values explicitly
8ce288b8f8 Update fiddler-base for 020ae52af0e7
ea25063ef6 Reset convexity in SkPathBuilder::addPath
dda360b5de Use Skia&apos;s ninja when building viewer with gradle
b51232c416 Add Louhi service account to AUTHORS
0116a44a6a Remove friend pathref hacking
c0bf040e6b [rust png] Copy `EncodePngAsSkData` from Chromium into Skia.
05c1f58034 Roll vulkan-deps from 29b917fb5921 to 3964dd7c126b (4 revisions)
707eef0fc3 Roll ANGLE from ae02c3292a95 to 7994bf76d7a9 (8 revisions)
48cd91858f Roll recipe dependencies (trivial).
69045151f8 Roll Skia Infra from a3b8ecf9f94a to 8d4953412be2 (6 revisions)
257c1f94af Roll Dawn from cd3a5de0811f to 603a5155599a (14 revisions)
c4d8b104a1 Make SkPathPriv::Raw handle SkPaths with infinite bounds correctly
5868d0cc08 Reapply &quot;Reapply &quot;Use pathbuilder or factories to keep path immutable&quot;&quot;
0988b4a901 [graphite] Make use of PipelineCreateCacheControl in Vulkan
817b6f8f74 Update tests to use pathbuilder
e9d38d65b6 [skcapture] add capture callbacks for SkSurface
b5d8ae8d34 Revert &quot;Reapply &quot;Use pathbuilder or factories to keep path immutable&quot;&quot;
adbf581d4e Roll vulkan-deps from 9023de589c86 to 29b917fb5921 (3 revisions)
ba8a690e2a Remove unused mutable-path parameter
717da7f78d [graphite] Add SupportsPipelineCreationCacheControl bool to VulkanCaps
c77cd71718 Only enable shader precision modifier usage for ganesh/GL
c44a36470d Roll ANGLE from 81cda6698311 to ae02c3292a95 (9 revisions)
bf13164dbf Roll Dawn from c2dcfa0865c5 to cd3a5de0811f (31 revisions)
769ec67846 Roll recipe dependencies (trivial).
bb35a79646 Roll Skia Infra from e521bec27829 to a3b8ecf9f94a (9 revisions)
ecaff95f51 Roll SwiftShader from 5f1c459a11bb to 18d4f3db9407 (1 revision)
6998b06397 Roll vulkan-deps from 02470a3c2773 to 9023de589c86 (6 revisions)
521b24236f Roll recipe dependencies (trivial).
b242cc0948 Fix unused var in gm/imagemakewithfilter.cpp
1ac531a83d [graphite] Use correct reuse policy on mtl/vk graphite buffers.
9a99262e53 Merge 6 release notes into RELEASE_NOTES.md
e6215bb3f9 Update Skia milestone to 143
a93ab1d466 Roll recipe dependencies (trivial).
cdf24b9ee8 Viewer builds with Graphite but not Ganesh
d26846c006 Add new Encode() variant to _none impls
3d2b08a88e [graphite] Make use of ThreadSafeResourceProvider
5d5b63e354 Roll vulkan-deps from c18184683c0b to 02470a3c2773 (1 revision)
2b6ee4ef60 Reorganize SkDraw::drawPath() to keep SkPath immutable
33a7dc6f15 [pdf] Remove no longer needed #undefs
90dadc349e Roll ANGLE from 7a7681cc8f88 to 81cda6698311 (9 revisions)
3f2f1349c4 Roll Dawn from d930194b7928 to c2dcfa0865c5 (20 revisions)
d02785a1df Roll Skia Infra from ed3dfff05a5e to e521bec27829 (5 revisions)
d0e46e713d Roll vulkan-deps from a9e2ca3b57ab to c18184683c0b (7 revisions)
973908df6e compare utility for paths
a1a6503393 Add new Encode api to NDK variants
821753c0a1 Reapply &quot;Use pathbuilder or factories to keep path immutable&quot;
5ec97ae774 Apply font variation arguments to system default fonts and character lookup fallback fonts in SkParagraph&apos;s FontCollection
16e6da5e6f Harden our conic math for bad W values
b1ae2340a1 [graphite] Add a ThreadSafeResourceProvider
4c7ae6ee94 Unify most common form of encoders: returning SkData
bb3b6bd4be Roll vulkan-deps from 666c7222bfe6 to a9e2ca3b57ab (2 revisions)
26aa24f093 Roll ANGLE from 1f40285c90cd to 7a7681cc8f88 (8 revisions)
0b565086d2 Roll recipe dependencies (trivial).
1c10d9bb7d Roll Skia Infra from 77be60075080 to ed3dfff05a5e (3 revisions)
9b2b942d1e Roll Dawn from a1b9b261f48e to d930194b7928 (2 revisions)
b1e89bda2b Roll vulkan-deps from 7984c7451ed6 to 666c7222bfe6 (1 revision)
70a4803c0b Roll vulkan-deps from fc82537b4a40 to 7984c7451ed6 (3 revisions)
beb6739688 Roll vulkan-deps from 14f79df6f67f to fc82537b4a40 (3 revisions)
2e5da5c0a9 Manual roll Dawn from 443e526bf00b to a1b9b261f48e (14 revisions)
fed805fe6e Roll vulkan-deps from 2458bba6f0d0 to 14f79df6f67f (6 revisions)
d8422aaf8f Avoid dlsym on Android 10
2b871d6b99 Roll vulkan-deps from acc63a089712 to 2458bba6f0d0 (5 revisions)
96b73f61fe Roll ANGLE from 69745720973d to 1f40285c90cd (4 revisions)
b54e1064cc Roll Dawn from a208d31bc037 to 443e526bf00b (17 revisions)
c514858a24 Roll Skia Infra from 9e5871e71d0c to 77be60075080 (6 revisions)
5d99c3fc7c Roll vulkan-deps from 109cf2d238b7 to acc63a089712 (7 revisions)
c7f371b53a Revert &quot;Use pathbuilder or factories to keep path immutable&quot;
0e11b83078 Disable SkRuntimeBlender* tests on Protected jobs
5df1b8dc7a Correct blend info in case of depth-only pipeline
55436d87e4 [graphite][vulkan] Support VK_ext_frame_boundary extension
e50361ed98 Roll vulkan-deps from 7c910be6a4f5 to 109cf2d238b7 (4 revisions)
8b78c4a4fd Roll ANGLE from b4d1333d1caf to 69745720973d (15 revisions)
d82eac335b Roll recipe dependencies (trivial).
5380d8a32f Roll Skia Infra from 568da2c9bd8c to 9e5871e71d0c (9 revisions)
e2e8a4a815 Roll Dawn from c15f5f328283 to a208d31bc037 (15 revisions)
affcd3c572 Roll recipe dependencies (trivial).
eab18034ee Roll recipe dependencies (trivial).
f6d0d9e3fe Roll recipe dependencies (trivial).
02e8d931a8 Use pathbuilder or factories to keep path immutable
a47e6acad5 SkPath::getLastPt() to return optional
83c55cac45 Add Graphite flavor of SkRuntimeBlender test
fd2aae026a Roll vulkan-deps from 38ebdaf28c68 to 7c910be6a4f5 (2 revisions)
61fe0dc0a9 [graphite] Move VkPipelineCache to VulkanSharedContext
b47d63d47e Roll recipe dependencies (trivial).
c21cebc1a5 Add API and TEST_API variables to recipe modules
753ce2221c Roll ANGLE from 096924c1e93f to b4d1333d1caf (30 revisions)
ba1ddfdc68 Roll SwiftShader from 7dd631d218ad to 5f1c459a11bb (1 revision)
86fc83e987 Roll Dawn from 67b75214b0b5 to c15f5f328283 (56 revisions)
c53fe14b03 Roll Skia Infra from d549c8c3724a to 568da2c9bd8c (28 revisions)
1c1b19f2ff Roll vulkan-deps from a8e5fdb8672c to 38ebdaf28c68 (6 revisions)
16ab3a013d Roll recipe dependencies (trivial).
3191a822cf [graphite] Move findOrCreateGraphicsPipeline to SharedContext
e411b39b70 Guards blur fast path from non rrect
842c0f4f33 Roll recipe dependencies (trivial).
4683c30404 Specify path to ninja for chromebook compile jobs
727d8b9419 Remove SkPngEncoder,SkJpegEncoder,SkWebpEncoder::Options::fICCProfile
f1fab32641 Use pathraw to create mask from path
a89da02f11 Roll recipe dependencies (trivial).
f2a06ae1f7 Roll vulkan-deps from 548681014416 to a8e5fdb8672c (13 revisions)
ee33533e6a Disable failing WASM perf/test jobs
144222d3bc Disable failing CanvasKit BuildStats jobs
dd795be142 Refactor build_tint.py code into shared file
47f75036dd Fix YCbCr unit test crash
128746041e pathops Simplify() to return its result
a2c99b39a7 Convert tests to use pathbuilder
cabeab8cb2 Roll recipe dependencies (trivial).
593c8eb07c [rustpng] Relax assert for setting sigBits
e8fdbb0c5d [rust png] Fixing forward implicit int cast.
0a28d8f186 [rust png] Avoid `covered-switch-default` warning in `SkEncodedInfo.h`.
69fedde544 More tools/ gms/ changes in anticipation of Graphite-only viewer build.
60d9f14b52 More return resulting path by optional-value for internal pathops
5aafacb135 Revert &quot;Reland &quot;[graphite] Extracts early in drawGeometry&quot;&quot;
f1b7ab22a4 Fix GrProgramDesc creation wrt dstReads
ef27842f47 Convert tests to use pathbuilder
e3b899e6c4 PathOps Op() return its result
81de4113e3 Reland &quot;[graphite] Extracts early in drawGeometry&quot;
aa2aa57b79 Don&apos;t run NotifyInUseTest* tests on Protected jobs
aca11a7d25 Roll vulkan-deps from 8ff085ff814f to 548681014416 (15 revisions)
8d5cb2e101 Convert gms to use pathbuilder
4b362dad8e Regenerate .gni files
441be63a92 [graphite] Add NotifyInUse* Tests
a38a531dec [graphite] Add task dumping
82156da9be Migrate gms to use pathbuilder
d2205a6fef Return paths as results, aiding our conversion to immutability.
1dae085e2f Revert &quot;[ganesh] Fix bounds for subpixel StrokeRectOps&quot;
b56003bf2c BitmapRegionDecoder: Enable AVIF support
5e9223ed18 SkPngRustDecoder: Add HDR metadata support
c74d2bdbd9 Roll ANGLE from f5f9c129a75c to 096924c1e93f (13 revisions)
7f2d54dc65 Roll vulkan-deps from 86b0a2e14a4d to 8ff085ff814f (2 revisions)
684f3a8312 Roll SwiftShader from ce5b85176a89 to 7dd631d218ad (1 revision)
82f56c712c Roll Skia Infra from 3d311dfa7fdd to d549c8c3724a (7 revisions)
38a42d53e2 Roll Dawn from d88a7a749566 to 67b75214b0b5 (6 revisions)
462bdece17 [rustpng] Read sBIT when creating SkEncodedInfo
d31a6ef0e4 [graphite] Add limit to number of paths added to small path atlas per flush
9d1eb22dd5 [ganesh] Fix bounds for subpixel StrokeRectOps
c9ffd2b0c1 Remove stale TODO comment -- bug is closed.
785a9bcb36 Roll vulkan-deps from 55678f5f6dfc to 86b0a2e14a4d (1 revision)
61fee81a0f [rust png] Fix `blendAllRows` to avoid `subspan`ing beyond the buffer.
6887dcf153 Patch wacky_yuv_formats_imggen GM
57f6a74914 Move interp helper into private headers
69d74e9d07 Start to migrate pathops to builder
0202b76b4d Manual roll skcms from 7852fc3f0085 to 96d9171c94b9 (1 revision)
8cc3f653a3 Migrate Ubuntu18 jobs onto Ubuntu24 machines
a2c38aa9df Roll ANGLE from c8af6941bc39 to f5f9c129a75c (13 revisions)
7b9fe91446 Roll Dawn from 7e9df7778942 to d88a7a749566 (6 revisions)
16c8c4d263 Roll Skia Infra from 6258ef4e4ef1 to 3d311dfa7fdd (12 revisions)
02d91221d5 Roll vulkan-deps from 88879365c796 to 55678f5f6dfc (8 revisions)
106bc3af44 Roll recipe dependencies (trivial).
ab1b105474 Roll recipe dependencies (trivial).
2d2b9c7142 [rustpng] Expand test coverage
d9b6dc608a Reland &quot;[rustpng] Implement subsets for SkPngRustCodec&quot;
d9e2467c31 [rust png] Deprecate old Bazel target names without explicit `libpng`.
63a6cbf423 Roll recipe dependencies (trivial).
8d12d8dc89 Use pathbuilder in path::transform + perspective
3745e62ff9 [rust png] Move Rust PNG implementation to non-&quot;experimental&quot; dir.
d058a4ab69 Manual roll SwiftShader from 3877515497c0 to ce5b85176a89 (1 revision)
f9d6a55c23 [graphite] Add multiple work lists to SkExecutor
7d56c17f64 Manual roll ANGLE from 3760c722c0c8 to c8af6941bc39 (20 revisions)
54f27fe79a [graphite] Avoid swizzling + indexing within same line in vertex shaders
84028c93b5 [graphite] Use helper for Vulkan ShaderCaps initialization
db91ae676e Revert &quot;[rustpng] Implement subsets for SkPngRustCodec&quot;
760a2a4b9d [graphite][vulkan]: handle explicit YUV VkFormat resolves
09bfdfa9d9 Remove valgrind from our test configurations
82de95a10c SkCodec/SkPngCodec/SkPngEncoder: Add HDR metadata support
2a3e9c7543 Manual roll Dawn from 8305e09d1c00 to 7e9df7778942 (18 revisions)
79ec8dfcd9 [graphite] Remove redundant naming within *Caps files
82fc2dc3a6 [graphite] Move findGraphicsPipeline to be entirely in GlobalCache
f60b25c165 Roll recipe dependencies (trivial).
0a90ff9056 Roll vulkan-deps from 19d614b75314 to 88879365c796 (5 revisions)
b4d3b8e27e [graphite] Move Dawn BindGroupLayouts from ResourceProvider to SharedContext
28f9ceaa26 Use pathbuilder to keep path immutable
a5a8f1df14 Fix Gofmt error
c69f623478 Assume LazyTexture images have successfully been pinned.
c897c35ef8 Roll recipe dependencies (trivial).
6da4a8b5c3 Convert gm to use pathbuilder
37252404fa Convert gms and tests to use pathbuilder
2c0b3930aa Remove references to retired hardware
283a8ff3bc Revert &quot;[graphite] Add multiple work lists to SkExecutor&quot;
e735ac6910 Pretty print jobs.json
bc7f3891e8 [rustpng] Implement subsets for SkPngRustCodec
add9179c50 Return results, so they can be immutable
76a371785e Refactor compute-convexity
3b83fd3eba [graphite] Add multiple work lists to SkExecutor
a44e5e2d95 [graphite] Move TokenTracker out of AtlasTypes
cf40d2210f [graphite] Move DepthStencilState caching to MtlSharedContext
c42d0f2707 Roll recipe dependencies (trivial).
475bdf3782 Roll recipe dependencies (trivial).
48181e2c2c Roll vulkan-deps from 603c3e033c87 to 19d614b75314 (6 revisions)
7d160bbf94 Roll ANGLE from 81a181d12531 to 3760c722c0c8 (7 revisions)
535e3fc629 Roll Skia Infra from 3f6aa4f56ad1 to 6258ef4e4ef1 (6 revisions)
a7b6e74a75 Roll Dawn from af2f85f5494c to 8305e09d1c00 (4 revisions)
4e9c86d4a6 skhdr::Metadata: Add CLLI and MDCV metadata and parsing
01b0ede33a [rust png] Move Rust PNG public headers to non-&quot;experimental&quot; dirs (3/3)
8fbb411c28 Roll vulkan-deps from fe358c043770 to 603c3e033c87 (5 revisions)
93ebee225f Use pathbuilder to keep paths immutable
f5c7bb1c5b Revert &quot;SkColorSpacePrimaries: Add operator== and operator!=&quot;
9121f335b4 Revert &quot;skhdr::Metadata: Add CLLI and MDCV metadata and parsing&quot;
d4b9dd2952 Roll recipe dependencies (trivial).
f2af25ed64 skhdr::Metadata: Add CLLI and MDCV metadata and parsing
f950263bb3 Make GMs compilable without Ganesh enabled
d84a369255 SkColorSpacePrimaries: Add operator== and operator!=

Canonical link: <a href="https://commits.webkit.org/301060@main">https://commits.webkit.org/301060@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f71fff69f96e769fb414c9c253ce8805b970a026

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/124847 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/44518 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/35254 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/131696 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/76752 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/126724 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/45220 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/53087 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/95004 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/76752 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/127801 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/36077 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/111655 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/75566 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/35011 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/29811 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/75169 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/116955 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/105836 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/136/builds/30040 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/134360 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/123367 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/51690 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/39495 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/103483 "") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/52106 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/107870 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/103243 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26286 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/48607 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/26894 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/48709 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/51565 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/156393 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/50955 "Built successfully") | | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/25/builds/39170 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/54311 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/52650 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/f71fff69f96e769fb414c9c253ce8805b970a026. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51861

aperezdc committed:
Update Skia to 2025-10-06 (ed728afeb3ff52689f260a6dfa23e31df1d93166)
https://bugs.webkit.org/show_bug.cgi?id=300228

Unreviewed. The changes are localized to Source/ThirdParty/skia.

% git log --oneline --pretty='%h %s' 953ef34aabaeda010e05c578a8e2de5c08e25c91..ed728afeb3ff52689f260a6dfa23e31df1d93166
ed728afeb3 Roll ANGLE from 62b00f866364 to fc98b3f62a98 (11 revisions)
45191c22b1 Roll Dawn from 01940842b667 to d8f0feef7c17 (14 revisions)
65347fdf4d Roll Skia Infra from 228c951bd699 to b57557664701 (8 revisions)
e7c94f0f38 Roll vulkan-deps from 207fff6486db to 7b14f3ac9637 (1 revision)
1fd0ca1f21 Roll SKP CIPD package from 533 to 534
5479115ef5 Roll vulkan-deps from a45aa5b3843e to 207fff6486db (1 revision)
35f8b9a341 [graphite] Add findOrCreateScratchBuffer to ResourceProvider
67489f2f45 [graphite] add missing include in VelloComputeSteps.h
3d6c568f17 Roll vulkan-deps from edacf5135c8d to a45aa5b3843e (8 revisions)
f316de3d47 Roll recipe dependencies (trivial).
81cab272e3 Return optional raw for path/builder
9cda1a2050 [graphite] Remove assert in VellComputeStep
8a7f88c5b3 [graphite] Move functionality into base BufferWriter
a454242c39 Reland "Reland "[graphite] Extracts early in drawGeometry""
eb27c53d93 [graphite] Add all blends to NotifyInUseTests
cb9fec0711 Roll vulkan-deps from 7bc87a4a45cf to edacf5135c8d (4 revisions)
b842026480 Roll ANGLE from 7994bf76d7a9 to 62b00f866364 (11 revisions)
3122894ae6 Roll recipe dependencies (trivial).
a256cc54c5 Roll Skia Infra from 8d4953412be2 to 228c951bd699 (11 revisions)
f86ae41132 Roll Dawn from 603a5155599a to 01940842b667 (22 revisions)
1720a85a50 Roll recipe dependencies (trivial).
c6effcb43b Roll vulkan-deps from 3964dd7c126b to 7bc87a4a45cf (7 revisions)
c7792a4a68 Roll recipe dependencies (trivial).
93e2cb6281 Add tests for SkSGMerge
cf339ab390 Make SkAlignTo templated; optimize SkAlignNonPow2
0d6067256c Roll recipe dependencies (trivial).
86f7e448c8 [graphite] Add drawContext pointer to task dumping
7a8c30bd76 use pathbuilder or factories, rather than mutate SkPath
c7f4bad47e Update pixmap asserts to identify negative values explicitly
8ce288b8f8 Update fiddler-base for 020ae52af0e7
ea25063ef6 Reset convexity in SkPathBuilder::addPath
dda360b5de Use Skia's ninja when building viewer with gradle
b51232c416 Add Louhi service account to AUTHORS
0116a44a6a Remove friend pathref hacking
c0bf040e6b [rust png] Copy `EncodePngAsSkData` from Chromium into Skia.
05c1f58034 Roll vulkan-deps from 29b917fb5921 to 3964dd7c126b (4 revisions)
707eef0fc3 Roll ANGLE from ae02c3292a95 to 7994bf76d7a9 (8 revisions)
48cd91858f Roll recipe dependencies (trivial).
69045151f8 Roll Skia Infra from a3b8ecf9f94a to 8d4953412be2 (6 revisions)
257c1f94af Roll Dawn from cd3a5de0811f to 603a5155599a (14 revisions)
c4d8b104a1 Make SkPathPriv::Raw handle SkPaths with infinite bounds correctly
5868d0cc08 Reapply "Reapply "Use pathbuilder or factories to keep path immutable""
0988b4a901 [graphite] Make use of PipelineCreateCacheControl in Vulkan
817b6f8f74 Update tests to use pathbuilder
e9d38d65b6 [skcapture] add capture callbacks for SkSurface
b5d8ae8d34 Revert "Reapply "Use pathbuilder or factories to keep path immutable""
adbf581d4e Roll vulkan-deps from 9023de589c86 to 29b917fb5921 (3 revisions)
ba8a690e2a Remove unused mutable-path parameter
717da7f78d [graphite] Add SupportsPipelineCreationCacheControl bool to VulkanCaps
c77cd71718 Only enable shader precision modifier usage for ganesh/GL
c44a36470d Roll ANGLE from 81cda6698311 to ae02c3292a95 (9 revisions)
bf13164dbf Roll Dawn from c2dcfa0865c5 to cd3a5de0811f (31 revisions)
769ec67846 Roll recipe dependencies (trivial).
bb35a79646 Roll Skia Infra from e521bec27829 to a3b8ecf9f94a (9 revisions)
ecaff95f51 Roll SwiftShader from 5f1c459a11bb to 18d4f3db9407 (1 revision)
6998b06397 Roll vulkan-deps from 02470a3c2773 to 9023de589c86 (6 revisions)
521b24236f Roll recipe dependencies (trivial).
b242cc0948 Fix unused var in gm/imagemakewithfilter.cpp
1ac531a83d [graphite] Use correct reuse policy on mtl/vk graphite buffers.
9a99262e53 Merge 6 release notes into RELEASE_NOTES.md
e6215bb3f9 Update Skia milestone to 143
a93ab1d466 Roll recipe dependencies (trivial).
cdf24b9ee8 Viewer builds with Graphite but not Ganesh
d26846c006 Add new Encode() variant to _none impls
3d2b08a88e [graphite] Make use of ThreadSafeResourceProvider
5d5b63e354 Roll vulkan-deps from c18184683c0b to 02470a3c2773 (1 revision)
2b6ee4ef60 Reorganize SkDraw::drawPath() to keep SkPath immutable
33a7dc6f15 [pdf] Remove no longer needed #undefs
90dadc349e Roll ANGLE from 7a7681cc8f88 to 81cda6698311 (9 revisions)
3f2f1349c4 Roll Dawn from d930194b7928 to c2dcfa0865c5 (20 revisions)
d02785a1df Roll Skia Infra from ed3dfff05a5e to e521bec27829 (5 revisions)
d0e46e713d Roll vulkan-deps from a9e2ca3b57ab to c18184683c0b (7 revisions)
973908df6e compare utility for paths
a1a6503393 Add new Encode api to NDK variants
821753c0a1 Reapply "Use pathbuilder or factories to keep path immutable"
5ec97ae774 Apply font variation arguments to system default fonts and character lookup fallback fonts in SkParagraph's FontCollection
16e6da5e6f Harden our conic math for bad W values
b1ae2340a1 [graphite] Add a ThreadSafeResourceProvider
4c7ae6ee94 Unify most common form of encoders: returning SkData
bb3b6bd4be Roll vulkan-deps from 666c7222bfe6 to a9e2ca3b57ab (2 revisions)
26aa24f093 Roll ANGLE from 1f40285c90cd to 7a7681cc8f88 (8 revisions)
0b565086d2 Roll recipe dependencies (trivial).
1c10d9bb7d Roll Skia Infra from 77be60075080 to ed3dfff05a5e (3 revisions)
9b2b942d1e Roll Dawn from a1b9b261f48e to d930194b7928 (2 revisions)
b1e89bda2b Roll vulkan-deps from 7984c7451ed6 to 666c7222bfe6 (1 revision)
70a4803c0b Roll vulkan-deps from fc82537b4a40 to 7984c7451ed6 (3 revisions)
beb6739688 Roll vulkan-deps from 14f79df6f67f to fc82537b4a40 (3 revisions)
2e5da5c0a9 Manual roll Dawn from 443e526bf00b to a1b9b261f48e (14 revisions)
fed805fe6e Roll vulkan-deps from 2458bba6f0d0 to 14f79df6f67f (6 revisions)
d8422aaf8f Avoid dlsym on Android 10
2b871d6b99 Roll vulkan-deps from acc63a089712 to 2458bba6f0d0 (5 revisions)
96b73f61fe Roll ANGLE from 69745720973d to 1f40285c90cd (4 revisions)
b54e1064cc Roll Dawn from a208d31bc037 to 443e526bf00b (17 revisions)
c514858a24 Roll Skia Infra from 9e5871e71d0c to 77be60075080 (6 revisions)
5d99c3fc7c Roll vulkan-deps from 109cf2d238b7 to acc63a089712 (7 revisions)
c7f371b53a Revert "Use pathbuilder or factories to keep path immutable"
0e11b83078 Disable SkRuntimeBlender* tests on Protected jobs
5df1b8dc7a Correct blend info in case of depth-only pipeline
55436d87e4 [graphite][vulkan] Support VK_ext_frame_boundary extension
e50361ed98 Roll vulkan-deps from 7c910be6a4f5 to 109cf2d238b7 (4 revisions)
8b78c4a4fd Roll ANGLE from b4d1333d1caf to 69745720973d (15 revisions)
d82eac335b Roll recipe dependencies (trivial).
5380d8a32f Roll Skia Infra from 568da2c9bd8c to 9e5871e71d0c (9 revisions)
e2e8a4a815 Roll Dawn from c15f5f328283 to a208d31bc037 (15 revisions)
affcd3c572 Roll recipe dependencies (trivial).
eab18034ee Roll recipe dependencies (trivial).
f6d0d9e3fe Roll recipe dependencies (trivial).
02e8d931a8 Use pathbuilder or factories to keep path immutable
a47e6acad5 SkPath::getLastPt() to return optional
83c55cac45 Add Graphite flavor of SkRuntimeBlender test
fd2aae026a Roll vulkan-deps from 38ebdaf28c68 to 7c910be6a4f5 (2 revisions)
61fe0dc0a9 [graphite] Move VkPipelineCache to VulkanSharedContext
b47d63d47e Roll recipe dependencies (trivial).
c21cebc1a5 Add API and TEST_API variables to recipe modules
753ce2221c Roll ANGLE from 096924c1e93f to b4d1333d1caf (30 revisions)
ba1ddfdc68 Roll SwiftShader from 7dd631d218ad to 5f1c459a11bb (1 revision)
86fc83e987 Roll Dawn from 67b75214b0b5 to c15f5f328283 (56 revisions)
c53fe14b03 Roll Skia Infra from d549c8c3724a to 568da2c9bd8c (28 revisions)
1c1b19f2ff Roll vulkan-deps from a8e5fdb8672c to 38ebdaf28c68 (6 revisions)
16ab3a013d Roll recipe dependencies (trivial).
3191a822cf [graphite] Move findOrCreateGraphicsPipeline to SharedContext
e411b39b70 Guards blur fast path from non rrect
842c0f4f33 Roll recipe dependencies (trivial).
4683c30404 Specify path to ninja for chromebook compile jobs
727d8b9419 Remove SkPngEncoder,SkJpegEncoder,SkWebpEncoder::Options::fICCProfile
f1fab32641 Use pathraw to create mask from path
a89da02f11 Roll recipe dependencies (trivial).
f2a06ae1f7 Roll vulkan-deps from 548681014416 to a8e5fdb8672c (13 revisions)
ee33533e6a Disable failing WASM perf/test jobs
144222d3bc Disable failing CanvasKit BuildStats jobs
dd795be142 Refactor build_tint.py code into shared file
47f75036dd Fix YCbCr unit test crash
128746041e pathops Simplify() to return its result
a2c99b39a7 Convert tests to use pathbuilder
cabeab8cb2 Roll recipe dependencies (trivial).
593c8eb07c [rustpng] Relax assert for setting sigBits
e8fdbb0c5d [rust png] Fixing forward implicit int cast.
0a28d8f186 [rust png] Avoid `covered-switch-default` warning in `SkEncodedInfo.h`.
69fedde544 More tools/ gms/ changes in anticipation of Graphite-only viewer build.
60d9f14b52 More return resulting path by optional-value for internal pathops
5aafacb135 Revert "Reland "[graphite] Extracts early in drawGeometry""
f1b7ab22a4 Fix GrProgramDesc creation wrt dstReads
ef27842f47 Convert tests to use pathbuilder
e3b899e6c4 PathOps Op() return its result
81de4113e3 Reland "[graphite] Extracts early in drawGeometry"
aa2aa57b79 Don't run NotifyInUseTest* tests on Protected jobs
aca11a7d25 Roll vulkan-deps from 8ff085ff814f to 548681014416 (15 revisions)
8d5cb2e101 Convert gms to use pathbuilder
4b362dad8e Regenerate .gni files
441be63a92 [graphite] Add NotifyInUse* Tests
a38a531dec [graphite] Add task dumping
82156da9be Migrate gms to use pathbuilder
d2205a6fef Return paths as results, aiding our conversion to immutability.
1dae085e2f Revert "[ganesh] Fix bounds for subpixel StrokeRectOps"
b56003bf2c BitmapRegionDecoder: Enable AVIF support
5e9223ed18 SkPngRustDecoder: Add HDR metadata support
c74d2bdbd9 Roll ANGLE from f5f9c129a75c to 096924c1e93f (13 revisions)
7f2d54dc65 Roll vulkan-deps from 86b0a2e14a4d to 8ff085ff814f (2 revisions)
684f3a8312 Roll SwiftShader from ce5b85176a89 to 7dd631d218ad (1 revision)
82f56c712c Roll Skia Infra from 3d311dfa7fdd to d549c8c3724a (7 revisions)
38a42d53e2 Roll Dawn from d88a7a749566 to 67b75214b0b5 (6 revisions)
462bdece17 [rustpng] Read sBIT when creating SkEncodedInfo
d31a6ef0e4 [graphite] Add limit to number of paths added to small path atlas per flush
9d1eb22dd5 [ganesh] Fix bounds for subpixel StrokeRectOps
c9ffd2b0c1 Remove stale TODO comment -- bug is closed.
785a9bcb36 Roll vulkan-deps from 55678f5f6dfc to 86b0a2e14a4d (1 revision)
61fee81a0f [rust png] Fix `blendAllRows` to avoid `subspan`ing beyond the buffer.
6887dcf153 Patch wacky_yuv_formats_imggen GM
57f6a74914 Move interp helper into private headers
69d74e9d07 Start to migrate pathops to builder
0202b76b4d Manual roll skcms from 7852fc3f0085 to 96d9171c94b9 (1 revision)
8cc3f653a3 Migrate Ubuntu18 jobs onto Ubuntu24 machines
a2c38aa9df Roll ANGLE from c8af6941bc39 to f5f9c129a75c (13 revisions)
7b9fe91446 Roll Dawn from 7e9df7778942 to d88a7a749566 (6 revisions)
16c8c4d263 Roll Skia Infra from 6258ef4e4ef1 to 3d311dfa7fdd (12 revisions)
02d91221d5 Roll vulkan-deps from 88879365c796 to 55678f5f6dfc (8 revisions)
106bc3af44 Roll recipe dependencies (trivial).
ab1b105474 Roll recipe dependencies (trivial).
2d2b9c7142 [rustpng] Expand test coverage
d9b6dc608a Reland "[rustpng] Implement subsets for SkPngRustCodec"
d9e2467c31 [rust png] Deprecate old Bazel target names without explicit `libpng`.
63a6cbf423 Roll recipe dependencies (trivial).
8d12d8dc89 Use pathbuilder in path::transform + perspective
3745e62ff9 [rust png] Move Rust PNG implementation to non-"experimental" dir.
d058a4ab69 Manual roll SwiftShader from 3877515497c0 to ce5b85176a89 (1 revision)
f9d6a55c23 [graphite] Add multiple work lists to SkExecutor
7d56c17f64 Manual roll ANGLE from 3760c722c0c8 to c8af6941bc39 (20 revisions)
54f27fe79a [graphite] Avoid swizzling + indexing within same line in vertex shaders
84028c93b5 [graphite] Use helper for Vulkan ShaderCaps initialization
db91ae676e Revert "[rustpng] Implement subsets for SkPngRustCodec"
760a2a4b9d [graphite][vulkan]: handle explicit YUV VkFormat resolves
09bfdfa9d9 Remove valgrind from our test configurations
82de95a10c SkCodec/SkPngCodec/SkPngEncoder: Add HDR metadata support
2a3e9c7543 Manual roll Dawn from 8305e09d1c00 to 7e9df7778942 (18 revisions)
79ec8dfcd9 [graphite] Remove redundant naming within *Caps files
82fc2dc3a6 [graphite] Move findGraphicsPipeline to be entirely in GlobalCache
f60b25c165 Roll recipe dependencies (trivial).
0a90ff9056 Roll vulkan-deps from 19d614b75314 to 88879365c796 (5 revisions)
b4d3b8e27e [graphite] Move Dawn BindGroupLayouts from ResourceProvider to SharedContext
28f9ceaa26 Use pathbuilder to keep path immutable
a5a8f1df14 Fix Gofmt error
c69f623478 Assume LazyTexture images have successfully been pinned.
c897c35ef8 Roll recipe dependencies (trivial).
6da4a8b5c3 Convert gm to use pathbuilder
37252404fa Convert gms and tests to use pathbuilder
2c0b3930aa Remove references to retired hardware
283a8ff3bc Revert "[graphite] Add multiple work lists to SkExecutor"
e735ac6910 Pretty print jobs.json
bc7f3891e8 [rustpng] Implement subsets for SkPngRustCodec
add9179c50 Return results, so they can be immutable
76a371785e Refactor compute-convexity
3b83fd3eba [graphite] Add multiple work lists to SkExecutor
a44e5e2d95 [graphite] Move TokenTracker out of AtlasTypes
cf40d2210f [graphite] Move DepthStencilState caching to MtlSharedContext
c42d0f2707 Roll recipe dependencies (trivial).
475bdf3782 Roll recipe dependencies (trivial).
48181e2c2c Roll vulkan-deps from 603c3e033c87 to 19d614b75314 (6 revisions)
7d160bbf94 Roll ANGLE from 81a181d12531 to 3760c722c0c8 (7 revisions)
535e3fc629 Roll Skia Infra from 3f6aa4f56ad1 to 6258ef4e4ef1 (6 revisions)
a7b6e74a75 Roll Dawn from af2f85f5494c to 8305e09d1c00 (4 revisions)
4e9c86d4a6 skhdr::Metadata: Add CLLI and MDCV metadata and parsing
01b0ede33a [rust png] Move Rust PNG public headers to non-"experimental" dirs (3/3)
8fbb411c28 Roll vulkan-deps from fe358c043770 to 603c3e033c87 (5 revisions)
93ebee225f Use pathbuilder to keep paths immutable
f5c7bb1c5b Revert "SkColorSpacePrimaries: Add operator== and operator!="
9121f335b4 Revert "skhdr::Metadata: Add CLLI and MDCV metadata and parsing"
d4b9dd2952 Roll recipe dependencies (trivial).
f2af25ed64 skhdr::Metadata: Add CLLI and MDCV metadata and parsing
f950263bb3 Make GMs compilable without Ganesh enabled
d84a369255 SkColorSpacePrimaries: Add operator== and operator!=

Canonical link: https://commits.webkit.org/301060@main
